jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostModel.scala

```diff
@@ -152,7 +151,7 @@ abstract class XGBoostModel(protected var _booster: Booster)
    * @param testSet test set represented as RDD
    * @param missingValue the specified value to represent the missing value
    */
-  def predict(testSet: RDD[MLDenseVector], missingValue: Float): RDD[Array[Array[Float]]] = {
+  def predict(testSet: RDD[MLDenseVector], missingValue: Float): RDD[Array[Float]] = {
```

Comment:
I thought about this more and I am not sure it is worth going through the deprecation cycle for this particular API, because

1. it does not satisfy the user expectations regarding the shape of the output;
2. the users are probably doing the same hack as the tests: `.collect()(i)`, which is easy to adjust to the new API;
3. there're no official releases of xgboost4j, no published JARs, nothing, therefore the deprecation cycle is hard to implement properly;
4. any other name than `predict` would be harder to guess and it's a shame to have it taken by the broken implementation.

What do you think?